From 9aab93507a7834bfef7fb0a5f709393bcd8e72fe Mon Sep 17 00:00:00 2001
From: Sairamreddy Bojja <quic_sbojja@quicinc.com>
Date: Mon, 19 May 2025 16:40:41 +0530
Subject: [PATCH] ar_osal: update msm_audio.h header include path.

for staging kenel msm_audio.h coming as part of uapi headers.
add staging kernel sanitized headers CFLAG.

Upstream-Status: Pending
Signed-off-by: Sairamreddy Bojja <quic_sbojja@quicinc.com>
---
 ar_osal/src/linux/qcom/ar_osal_shmem_db.c | 2 +-
 configure.ac                              | 5 +++++
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/ar_osal/src/linux/qcom/ar_osal_shmem_db.c b/ar_osal/src/linux/qcom/ar_osal_shmem_db.c
index 0b3b66e..4214cb5 100644
--- a/ar_osal/src/linux/qcom/ar_osal_shmem_db.c
+++ b/ar_osal/src/linux/qcom/ar_osal_shmem_db.c
@@ -20,7 +20,7 @@
 #include <sys/ioctl.h>
 #include <linux/dma-buf.h>
 #include <linux/dma-heap.h>
-#include <linux/msm_audio.h>
+#include <msm_audio.h>
 #ifdef AR_OSAL_USE_CUTILS
 #include <cutils/properties.h>
 #endif
diff --git a/configure.ac b/configure.ac
index 289784e..9ab920b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -46,6 +46,11 @@ fi
 
 AM_CONDITIONAL(USE_GLIB, test "x${with_glib}" = "xyes")
 
+AC_ARG_WITH(sanitized-headers,
+      AS_HELP_STRING([--with-sanitized-headers=DIR],
+         [Specify the location of the sanitized Linux headers]),
+         [CFLAGS="$CFLAGS -I$withval"])
+
 AC_ARG_WITH([cutils],
     AS_HELP_STRING([Use cutils (default is yes)]),
      [with_cutils=$withval],
-- 
2.34.1

